### DISCUSSION TITLE
Sign in (get token) from embedded client

### DISCUSSION THREAD

**Discussion by joakimgk** [supabase/supabase#11578] [Questions] [answered]
I'm using rest/https calls to post data to a Supabase table from an ESP8266 sensor, which is programmed in C++. As such, I don't have any Supabase library or client, only a https client. How can I then have the sensor sign in, and receive a bearer token for use when posting data (when RLS is enabled)?

Should I create a "service user" for my sensor, which could then be given access to post to certain tables (using policies)? Would it be possible to use some simple username/password login for the sensor to use to authenticate when booting up?

Alternatively, I'm trying to set up policies which allow anonymous post to these tables, such that the sensor could add data -- perhaps by setting some "password hash" or some other rudimentary auth in the https header.

Any advice is greatly appreciated!

Joakim

**Comment by GaryAustin1** [comment, marked as answer]
You can send custom headers like your hash idea and use them in RLS/SQL with https://postgrest.org/en/stable/api.html#http-context.
I would have a hash to "id" table so you can change the hash for a device, but keep its existing data if the need arises.

You could also "signIn" a device with normal software signup and then use the UUID in a long lived JWT you mint and load into the device.
You could also add refresh token processing to that and shorter expire on the JWT.

And you could certainly sign in with a fake email and password, but would still need to deal with refreshing tokens to log or have longed a long lived jwt.  Not sure that is much better than the other options.

**Reply by GaryAustin1** [reply]
auth/gotrue is a REST API and is documented here. https://github.com/supabase/gotrue

**Comment by joakimgk** [comment]
Hi Gary, and thanks for helping (again!) :-)
Could you please give a link to the software signin method, I couldn't find this. Is it a http method to call which returns a token?

All docs seem to require using the Supabase library, and there isn't one for C++
